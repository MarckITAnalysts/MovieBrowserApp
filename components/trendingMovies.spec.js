import React from 'react';
import renderer from 'react-test-renderer';
import TrendingMovies from './trendingMovies';

jest.mock('react-native-snap-carousel', () => 'Carousel');

const mockedUsedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => mockedUsedNavigate,
}));

const data = [
  {
    "adult": false,
    "backdrop_path": "/pWsD91G2R1Da3AKM3ymr3UoIfRb.jpg",
    "genre_ids": [
      878,
      28,
      18
    ],
    "id": 933131,
    "original_language": "ko",
    "original_title": "황야",
    "overview": "After a deadly earthquake turns Seoul into a lawless badland, a fearless huntsman springs into action to rescue a teenager abducted by a mad doctor.",
    "popularity": 2140.707,
    "poster_path": "/zVMyvNowgbsBAL6O6esWfRpAcOb.jpg",
    "release_date": "2024-01-26",
    "title": "Badland Hunters",
    "video": false,
    "vote_average": 6.742,
    "vote_count": 433
  },
]

describe('Home screen test cases', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TrendingMovies data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
