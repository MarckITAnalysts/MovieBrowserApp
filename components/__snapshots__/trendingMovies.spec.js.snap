// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen test cases renders correctly 1`] = `
<View
  style={
    [
      {
        "marginBottom": 32,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#fff",
        },
        {
          "fontSize": 20,
          "lineHeight": 28,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "marginBottom": 20,
        },
        {
          "color": "white",
          "fontSize": 20,
          "marginBottom": 20,
          "marginHorizontal": 16,
        },
      ]
    }
  >
    Popular
  </Text>
  <Carousel
    data={
      [
        {
          "adult": false,
          "backdrop_path": "/pWsD91G2R1Da3AKM3ymr3UoIfRb.jpg",
          "genre_ids": [
            878,
            28,
            18,
          ],
          "id": 933131,
          "original_language": "ko",
          "original_title": "황야",
          "overview": "After a deadly earthquake turns Seoul into a lawless badland, a fearless huntsman springs into action to rescue a teenager abducted by a mad doctor.",
          "popularity": 2140.707,
          "poster_path": "/zVMyvNowgbsBAL6O6esWfRpAcOb.jpg",
          "release_date": "2024-01-26",
          "title": "Badland Hunters",
          "video": false,
          "vote_average": 6.742,
          "vote_count": 433,
        },
      ]
    }
    firstItem={1}
    inactiveSlideOpacity={0.6}
    itemWidth={465}
    renderItem={[Function]}
    slideStyle={
      {
        "alignItems": "center",
        "display": "flex",
      }
    }
    sliderWidth={750}
  />
</View>
`;
